### PR TITLE
Aws ecs

### PR DESCRIPTION
**Pull request by amitlimaye**
#### Description
*Changes proposed in this pull request.*

#### Test plan
*Outline the test plan used to test this change before merging it.*

> Fixes #.

### CODE REVIEW

File: constants/constants.go [hunk 1]
```diff
@@ -67,3 +67,16 @@ const (
...
+type DockerMonitorMode int
+
+const (
+	// DockerMode is a mode for docker monitor when trireme is running on host with just a docker daemon
+	DockerMode DockerMonitorMode = iota
+
+	// KubernetesMode is a mode for docker monitor when trireme is running on host which is part of a kubernetes cluster
+	KubernetesMode
+
+	// ECSMode is a mode for docker monitor when trireme is running on host which is part of ECS/AWS cluster
```
Comment (dstiliadis) [inline constants/constants.go]: I think that from a trireme standpoint this is essentially a NoProxyMode and not ECS mode. The ECS aspects are covered on the metadata extractors. We should change the naming convention here. We should also provide a default metadata extractor for ECS. 

In the whole PR for Trireme, nothing is ECS specific. All references to ECS should be removed and changed to NoProxyMode (or something across these lines).

File: internal/monitor/instance/docker/monitor.go [hunk 2]
```diff
@@ -180,6 +193,7 @@ func DefaultConfig() *Config {
 		SocketAddress:              constants.DefaultDockerSocket,
 		SyncAtStart:                true,
 		KillContainerOnPolicyError: false,
+		ECS: false,
```
Comment (dstiliadis) [inline internal/monitor/instance/docker/monitor.go]: See above. Nothing is ECS specific here. It is just a different mode.

File: internal/monitor/instance/docker/monitor.go [hunk 3]
```diff
@@ -217,6 +230,8 @@ type dockerMonitor struct {
 	// killContainerError if enabled kills the container if a policy setting resulted in an error.
 	killContainerOnPolicyError bool
 	syncAtStart                bool
+	ECS                        bool
+	cstore                     contextstore.ContextStore
```
Comment (dstiliadis) [inline internal/monitor/instance/docker/monitor.go]: same

Comment (amitlimaye) [reply internal/monitor/instance/docker/monitor.go]: Renamed ECS to NoProxyMode

File: internal/monitor/instance/docker/monitor.go (hunk 2 above)
Comment (amitlimaye) [reply internal/monitor/instance/docker/monitor.go]: Renamed ECS to NoProxyMode

File: constants/constants.go (hunk 1 above)
Comment (amitlimaye) [reply constants/constants.go]: Renamed ECS to NoProxyMode

File: internal/monitor/instance/docker/monitor.go [hunk 4]
```diff
@@ -628,8 +690,8 @@ func (d *dockerMonitor) startDockerContainer(dockerInfo *types.ContainerJSON) er
 
 		event = tevents.EventStart
 	}
-
-	if err := d.config.PUHandler.HandlePUEvent(contextID, event); err != nil {
+	zap.L().Info("Changing Container State to:", zap.String("Event", string(event)))
```
Comment (satyamsi) [inline internal/monitor/instance/docker/monitor.go]: remove

File: internal/monitor/instance/docker/monitor.go [hunk 5]
```diff
@@ -643,12 +705,17 @@ func (d *dockerMonitor) startDockerContainer(dockerInfo *types.ContainerJSON) er
 	}
 
 	if dockerInfo.HostConfig.NetworkMode == constants.DockerHostMode {
-		if err := d.setupHostMode(contextID, runtimeInfo, dockerInfo); err != nil {
+		if err = d.setupHostMode(contextID, runtimeInfo, dockerInfo); err != nil {
 			return fmt.Errorf("unable to setup host mode for container %s: %s", contextID, err)
 		}
+		//dockerInfo.Config.Labels["$id"]
```
Comment (satyamsi) [inline internal/monitor/instance/docker/monitor.go]: remove.

**Pull request merged**